src/main/java/com/squareup/javapoet/AnnotationSpec.java

```diff
@@ -99,6 +104,17 @@ public static Builder builder(Class<?> type) {
     return builder(ClassName.get(type));
   }
 
+  public static AnnotationSpec get(AnnotationMirror annotation) {
+    AnnotationSpec.Builder builder = AnnotationSpec.builder(
+        ClassName.get((TypeElement) annotation.getAnnotationType().asElement()));
+    for (Map.Entry<? extends ExecutableElement, ? extends AnnotationValue> entry
+        : annotation.getElementValues().entrySet()) {
+      String name = entry.getKey().getSimpleName().toString();
+      builder.addMember(name, CodeBlock.builder().add(entry.getValue().toString()).build());
```

Comment:
Does this work for all annotation value types? Would be useful to use Compile-Testing to exercise this.